modules/middleware/repo.go

```diff
@@ -345,6 +353,19 @@ func RepoAssignment(redirect bool, args ...bool) macaron.Handler {
 
 		ctx.Data["BranchName"] = ctx.Repo.BranchName
 		ctx.Data["CommitId"] = ctx.Repo.CommitId
+
+		userAgent := ctx.Req.Header.Get("User-Agent")
+		ua := user_agent.New(userAgent);
+		browserName, browserVersion := ua.Browser()
+		browserVersion = version.Normalize(browserVersion)
+
+		var chromeSupportsCopy bool
+    var firefoxSupportsCopy bool
+    chromeSupportsCopy = browserName == "Chrome" && version.Compare(browserVersion, CHROMECOPYSUPPORT, ">=")
+		firefoxSupportsCopy = browserName == "Firefox" && version.Compare(browserVersion, FIREFOXCOPYSUPPORT, ">=")
+
+		ctx.Data["BrowserSupportsCopy"] = chromeSupportsCopy || firefoxSupportsCopy
```

Comment:
```go
if (browserName == "Chrome" && version.Compare(browserVersion, CHROMECOPYSUPPORT, ">=")) ||
    (browserName == "Firefox" && version.Compare(browserVersion, FIREFOXCOPYSUPPORT, ">=")) {
    ctx.Data["BrowserSupportsCopy"] = true
}
```

Is it better or you prefer current one? Because variable `chromeSupportsCopy` and `firefoxSupportsCopy` only be used once.